### PR TITLE
Add hetrogenous network for azure and aws.

### PR DESCRIPTION
**Pull request by dvalinrh**

### CODE REVIEW

**Review by sousinha97 — APPROVED** [review]

File: ansible_roles/roles/azure_create/tasks/main.yml [hunk 1]
```diff
@@ -126,7 +134,7 @@
 - name: Report creation time
   lineinfile:
     path: "{{ working_dir }}/cloud_timings"
-    line: "instance start_time: {{ (azure_create_time_end.stdout) | int - (azure_create_time_start.stdout) | int }}"
+    line: "instance start_time: {{ (azure_create_time_start.stdout) | int - (azure_create_time_start.stdout) | int }}"
```
Comment (frival) [inline ansible_roles/roles/azure_create/tasks/main.yml:137]: That can't be right...

File: bin/burden [hunk 2]
```diff
@@ -1958,6 +1956,27 @@ verify_java_version()
...
+	entry=$1
+
+	echo bin/parse_net_line \"${entry}\"
+	networks_string=`bin/parse_net_line \"${entry}\"`
+	entries=`echo $networks_string | wc -w`
+	if [ $entries -lt 2 ]; then
+		cleanup_and_exit "$entry\nYou designated networks, but did not provide any extra system information.  Please resolve, aborting." 1
+	fi
+	gl_cloud_net_type=`echo $networks_string | cut -d' ' -f1`
+	echo here we are
```
Comment (frival) [inline bin/burden:1970]: Debug cruft

File: bin/burden [hunk 3]
```diff
@@ -1958,6 +1956,27 @@ verify_java_version()
...
+	if [[ $entry == *"Number"* ]]; then
+		echo here we are1
```
Comment (frival) [inline bin/burden:1972]: Also debug cruft

File: bin/burden (hunk 2 above)
Comment (frival) [inline bin/burden:1970]: Debug cruft

File: bin/burden [hunk 4]
```diff
@@ -2819,9 +2833,10 @@ usage()
 		echo "             number: How many disks to create and attach"
 		echo "             size: How large is the disk in gigabytes"
 		echo "             type: Type of disk creating"
-		echo "          <Networks;number=n;type=x"
-		echo "             number_networks: number of internal networks to create. Each network will"
-		echo "             create another system to connect to."
+		echo "          <Networks;<system>;<system>;....;type=x"
+		echo "             <system>: in case of cloud systems, it is the VM to use.  In case of local systems, it will"
+		echo "             the name of the system working with.  In the case of the cloud, if the second system"
```
Comment (frival) [inline bin/burden:2838]: I think this line needs to start with "be"

File: bin/burden [hunk 5]
```diff
@@ -2834,11 +2849,9 @@ usage()
...
 		echo "           m5.24xlarge:Disks;number=8;size=1200;type=gp2"
-		echo "         Example 3: Designate m5.24xlarge with 4 networks"
-		echo "           m5.24xlarge:Networks;number=4"
-		echo "         Example 4: Designate m5.24xlarge with sys tunings udp_fix and none"
+		echo "         Example 3: Designate m5.24xlarge with sys tunings udp_fix and none"
 		echo "           m5.24xlarge:Sysctl_settings=none+udp_fix"
-		echo "         Example 5: Designate m5.xlarge to be created in us-east-1 and zone b"
+		echo "         Example 4: Designate m5.xlarge to be created in us-east-1 and zone b"
 		echo "           \"m5.xlarge[region=us-east-1&zone=b]\""
 	fi
```
Comment (frival) [inline bin/burden:2856]: Please add an example for the heterogeneous networks case.

**Review by frival — CHANGES_REQUESTED** [review]
Also assuming this has been tested with no network, heterogeneous network, and homogeneous network.  But you know what happens when you assume...

**Pull request closed without merging**